src/http2_connection.erl

```diff
@@ -683,10 +668,9 @@ route_frame(
       stream_id=StreamId,
       type=?RST_STREAM
       },
-   #rst_stream{
-      error_code=EC
-      }},
+   Payload},
   #connection{} = Conn) ->
+    EC = http2_frame_rst_stream:error_code(Payload),
```

Comment:
What it cleans up is the clarity of the function clauses. I got into the bad habit of binding everything I would need in the pattern match, but really I needed to be patterning matching on decision points, and all the bindings could be handled later.